src/main/java/mainclasses/TaskList.java

```diff
@@ -0,0 +1,126 @@
+            String description = userInput.substring(userInput.indexOf("deadline") + 9, userInput.indexOf("/by") - 1);
+            String deadlineDate = userInput.substring(userInput.indexOf("/by") + 4, userInput.indexOf("/by") + 14);
+            LocalDate date = LocalDate.parse(deadlineDate, formatter);
+            String deadLineTime = userInput.substring(userInput.indexOf("/by") + 15);
+            LocalTime time = LocalTime.parse(deadLineTime, timeFormatter);
+            task = new Deadline(description, date, time);
+        } else {
+            String description = userInput.substring(userInput.indexOf("event") + 6, userInput.indexOf("/at") - 1);
+            String eventDate = userInput.substring(userInput.indexOf("/at") + 4, userInput.indexOf("/at") + 14);
+            LocalDate date = LocalDate.parse(eventDate, formatter);
+            String deadLineTime = userInput.substring(userInput.indexOf("/at") + 15);
+            LocalTime time = LocalTime.parse(deadLineTime, timeFormatter);
+            task = new Event(description, date, time);
+        }
+        newStorage.add(task);
+        return ("Got it. I've added this task: \n") + printTaskWithNoNum(task) + "\n"
+                + ("Now you have " + this.newStorage.size() + " tasks in the list.");
+    }
+
+    /**
+     * List out of the task that are in the task list storage
+     * @param tasks List of tasks
+     */
+    public String listTask(List<Task> tasks) {
+        String newListString = "";
+        for (int i = 0; i < tasks.size(); i++) {
+            newListString += (printTask(i, tasks.get(i)) + "\n");
+        }
+        return ("Here are the tasks in your list: \n") + newListString;
+    }
+
+    /**
+     * Set a specific task as completed based on the index of task in task list storage
+     * @param value int index of the task in list
+     */
+    public String setTaskAsDone(int value) {
+        this.newStorage.get(value).setDone(true);
+        return ("Nice! I've marked this task as done: \n") + printTask(value, newStorage.get(value));
+    }
+
+    /**
+     * Deletes a task based on string input of user
+     * @param userInput user input in string text
+     */
+    public String deleteTask(String userInput) {
+        String[] inputBreakdown = userInput.split(" ");
+        int taskToBeDeleted = Integer.valueOf(inputBreakdown[1]) - 1;
+        Task task = this.newStorage.get(taskToBeDeleted);
+        this.newStorage.remove(taskToBeDeleted);
+        return ("Noted. I've removed this task: \n") + printTaskWithNoNum(task) + "\n"
+                + ("Now you have " + this.newStorage.size() + " tasks in the list.");
+    }
+
+    public List<Task> getNewStorage() {
+        return newStorage;
+    }
+
+    /**
+     * Finds Tasks which description contains a certain keyword
+     * @param keyWord This is the keyword used to filter the list of tasks
+     */
+    public String findTasksWithKeyword(String keyWord) {
```

Comment:
Very minor point here. Will it be better to change "keyWord" to "keyword" as it is a single word?